src/XMakeBuildEngine/UnitTestsPublicOM/Construction/ConstructionEditing_Tests.cs

```diff
@@ -3199,6 +3199,62 @@ private static string ComposeExpectedProjectString(string expectedItem)
             return expected;
         }
 
+        /// <summary>
+        /// Add a property to an empty project
+        /// should add to new property group
+        /// </summary>
+        [Fact]
+        public void AddProperty_WithSdk_KeepsSdkAndImplicitImports()
+        {
+            var testSdkRoot = Path.Combine(Path.GetTempPath(), "MSBuildUnitTest");
+            var testSdkDirectory = Path.Combine(testSdkRoot, "MSBuildUnitTestSdk");
+
+            try
+            {
+                Directory.CreateDirectory(testSdkDirectory);
+
+                string sdkPropsPath = Path.Combine(testSdkDirectory, "Sdk.props");
```

Comment:
I think this needs to be `Path.Combine(testSdkDirectory, "Sdk", "Sdk.props")` to match the path where the parser is going to look for the Sdk imports.